### ISSUE TITLE
Plugin: update to new Transform API

### ISSUE THREAD

**Issue by andrew-ld** [objectbox/objectbox-java#1078]
```
WARNING:API 'android.registerTransform' is obsolete.
It will be removed in version 8.0 of the Android Gradle plugin.
The Transform API is removed to improve build performance. Projects that use the
Transform API force the Android Gradle plugin to use a less optimized flow for the
build that can result in large regressions in build times. It’s also difficult to
use the Transform API and combine it with other Gradle features; the replacement
APIs aim to make it easier to extend the build without introducing performance or
correctness issues.
```

this message is visible using objectbox-gradle-plugin 3.1.2 on android gradle plugin 7.2.0 running on gradle 7.3.3

**Comment by greenrobot-team** [comment]
Thanks for reporting! We are already aware of this.

Note that the transformer continues to work with Android Plugin 7.2.0 (except for #1077).

**Comment by AndroidDeveloperLB** [comment]
What are we supposed to do when we see this weird warning?

**Comment by greenrobot-team** [comment]
@AndroidDeveloperLB Nothing at the moment. It's a deprecation warning so everything should work as expected (again except for #1077).

We'll need to ship an updated Gradle plugin that uses the new APIs for this warning to go away.

**Comment by AndroidDeveloperLB** [comment]
@greenrobot-team I mean in general, whether we use the current repository or not. I can see it on projects that don't use it.

**Comment by greenrobot-team** [comment]
This is resolved with the just released version `3.3.1` which uses the new transform APIs.
